DSCResources/MSFT_xWebSiteDefaults/MSFT_xWebSiteDefaults.psm1

```diff
@@ -0,0 +1,176 @@
+                                    LogDirectory = (GetValue "siteDefaults/logFile" "directory")}    
+	return $getTargetResourceResult
+}
+
+######################################################################################
+# The Set-TargetResource cmdlet.
+# This function will change a default setting if not already set
+######################################################################################
+function Set-TargetResource
+{
+	param
+	(	
+        [ValidateSet("Machine")]
+        [parameter(Mandatory = $true)]
+        [string]$ApplyTo,
+        [ValidateSet("W3C","IIS","NCSA","Custom")]
+        [string]$LogFormat,
+        [string]$LogDirectory,
+        [string]$TraceLogDirectory,
+        [string]$DefaultApplicationPool,
+        [ValidateSet("true","false")]
+        [string]$AllowSubDirConfig
+    )
+
+        CheckIISPoshModule
+
+        SetValue "siteDefaults/logFile" "logFormat" $LogFormat
+        SetValue "siteDefaults/logFile" "directory" $LogDirectory
+        SetValue "siteDefaults/traceFailedRequestsLogging" "directory" $TraceLogDirectory
+        SetValue "applicationDefaults" "applicationPool" $DefaultApplicationPool
+        SetValue "virtualDirectoryDefaults" "allowSubDirConfig" $AllowSubDirConfig
+}
+
+######################################################################################
+# The Test-TargetResource cmdlet.
+# This will test whether all given values are already set in the current configuration
+######################################################################################
+function Test-TargetResource
+{
+    [OutputType([System.Boolean])]
+	param
+	(	
+        [ValidateSet("Machine")]
+        [parameter(Mandatory = $true)]
+        [string]$ApplyTo,
+        [ValidateSet("W3C","IIS","NCSA","Custom")]
+        [string]$LogFormat,
+        [string]$LogDirectory,
+        [string]$TraceLogDirectory,
+        [string]$DefaultApplicationPool,
+        [ValidateSet("true","false")]
+        [string]$AllowSubDirConfig
+	)
+
+    [bool]$DesiredConfigurationMatch = $true;
+
+    CheckIISPoshModule
+
+    $DesiredConfigurationMatch = CheckValue "virtualDirectoryDefaults" "allowSubDirConfig" $AllowSubDirConfig
+    if (!($DesiredConfigurationMatch)) { return $false }
+
+    $DesiredConfigurationMatch = CheckValue "siteDefaults/logFile" "logFormat" $LogFormat
+    if (!($DesiredConfigurationMatch)) { return $false }
+
+    $DesiredConfigurationMatch = CheckValue "siteDefaults/logFile" "directory" $LogDirectory
+    if (!($DesiredConfigurationMatch)) { return $false }
+
+    $DesiredConfigurationMatch = CheckValue "siteDefaults/traceFailedRequestsLogging" "directory" $TraceLogDirectory
+    if (!($DesiredConfigurationMatch)) { return $false }
+
+    $DesiredConfigurationMatch = CheckValue "applicationDefaults" "applicationPool" $DefaultApplicationPool
+    if (!($DesiredConfigurationMatch)) { return $false }
+    
+	return $DesiredConfigurationMatch
```

Comment:
If you go with my above comments I would just have this return $true as by this point the desired configuration is being matched.